bin/migration/tables/portalaccess.py

```diff
@@ -0,0 +1,67 @@
+from .helpers import check_existing_record, parse_to_timestamp, audit_entry_creation, log_failed_imports
+from datetime import datetime
+import uuid
+
+class PortalAccessManager:
+    def __init__(self, source_cursor):
+        self.source_cursor = source_cursor
+        self.failed_imports = set()
+
+    def get_data(self):
+        self.source_cursor.execute("SELECT * FROM public.users WHERE prerole = 'Level 3'")
+        return self.source_cursor.fetchall()
+
+    def migrate_data(self, destination_cursor, source_user_data):
+        batch_portal_user_data = []
+
+        for user in source_user_data:
+            user_id = user[0]
+
+            if not check_existing_record(destination_cursor,'portal_access', 'user_id', user_id):
+                id=str(uuid.uuid4())
+                password = 'password' # ?
+                status = 'ACTIVE' # ? 
+
+                last_access = datetime.now() # ?
+                invitation_datetime = datetime.now() # ?
+                registered_datetime = datetime.now() # ?
+                status = 'ACTIVE' # ? 
+                modified_at =parse_to_timestamp(user[17])
+                created_by = user[14]
+                created_at = parse_to_timestamp(user[15])
+
```

Comment:
@mazzopardi2  ‘Password’ field displaying the value as ‘password’. Do we need this field? What is the actual value going to be populated here?

Reply:
I believe this is a temporary field to be used until we get Azure B2C implemented. It's not data that we will be migrating from our current db. @lucas-phillips28 - I think you may know more about this field?